File: cpp/Program/Clothing.cpp
#include <bits/stdc++.h>
#include "Product.cpp"

using namespace std;

class Clothing: public Product{
    private:
        string size;
        string material;
        char gender;
    
    public:
        Clothing(){
            this->size = "";
            this->material = "";
            this->gender = '-';
        }

        Clothing(string size, string material, char gender){
            this->size = size;
            this->material = material;
            this->gender = gender;
        }
        
        // setter dan getter
        string getSize(){
            return this->size;
        }

        void setSize(string size){
            this->size = size;
        }

        string getMaterial(){
            return this->material;
        }

        void setMaterial(string material){
            this->material = material;
        }

        char getGender(){
            return this->gender;
        }

        void setGender(char gender){
            this->gender = gender;
        }

        ~Clothing(){
            
        }
};

File: java/Program/Clothing.java
class Clothing extends Product{ // anak dari kelas Product
    // variabel private
    private String size;
    private String material;
    private char gender;

    // constructor
    Clothing(){
        size = "";
        material = "";
        gender = '-';
    }

    Clothing(String size, String material, char gender){
        this.size = size;
        this.material = material;
        this.gender = gender;
    }

    // setter dan getter
    public String getSize(){
        return this.size;
    }

    public void setSize(String size){
        this.size = size;
    }

    public String getMaterial(){
        return this.material;
    }

    public void setMaterial(String material){
        this.material = material;
    }

    public char getGender(){
        return this.gender;
    }

    public void setGender(char gender){
        this.gender = gender;
    }
}